### PR TITLE
CDAP-1255 the explore service was not returning any results

### PR DESCRIPTION
**Pull request by albertshau**
because it was translating a RowSet to a TRowSet (thrift object),
and then getting the rows from that. But different versions of
the protocol get the rows differently, with older versions using
a RowBasedSet and newer versions using a ColumnBasedSet that
behave differently. Instead, we should just be using the RowSet
directly.

### CODE REVIEW

**Comment by albertshau** [comment]
tested on hdp2.4 and cdh5.3.  Will test on cdh5.0 and cdh4.3 as well

**Comment by albertshau** [comment]
https://builds.cask.co/browse/CDAP-RBT109

File: cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/BaseHiveExploreService.java [hunk 1]
```diff
@@ -676,35 +671,19 @@ public QueryStatus getStatus(QueryHandle handle) throws ExploreException, Handle
...
-        ImmutableList.Builder<QueryResult> rowsBuilder = ImmutableList.builder();
-        for (TRow tRow : tRowSet.getRows()) {
-          List<Object> cols = Lists.newArrayList();
-          for (TColumnValue tColumnValue : tRow.getColVals()) {
-            cols.add(tColumnToObject(tColumnValue));
-          }
-          rowsBuilder.add(new QueryResult(cols));
+    if (operationHandle.hasResultSet()) {
+      ImmutableList.Builder<QueryResult> rowsBuilder = ImmutableList.builder();
+      Iterable<Object[]> rowSet = getCliService().fetchResults(operationHandle, FetchOrientation.FETCH_NEXT, size);
```
Comment (jguery) [inline cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/BaseHiveExploreService.java]: So in every cases, in all Hive versions, the result of ``fetchResults`` is always an ``Iterable<Object[]>`` ?

**Comment by jguery** [comment]
I'm just concerned that not all versions of Hive return a ``Iterable<Object[]>``. But if that's the case, and we don't need reflection anymore, then LGTM by all means!

**Comment by albertshau** [comment]
ugh ok just saw that its not in an old version

**Comment by jguery** [comment]
:/

**Comment by albertshau** [comment]
yeah cdh4.x

**Comment by jguery** [comment]
Looks good to me, if the tests on cdh4.3 work with that.

**Comment by albertshau** [comment]
tested on cdh4.x,cdh5.0,cdh5.3, hdp2.4. Will wait for unit tests before merging

**Pull request merged**